docs/decisions/20240103-use-jwt-claims-for-agent-admin-auth.md

```diff
@@ -0,0 +1,155 @@
+# Use JWT claims for agent admin access control
+
+- Status: accepted
+- Deciders: Pat Losoponkul, Yurii Shynbuiev, David Poltorak, Shailesh Patil
+- Date: 2024-01-03
+- Tags: multitenancy, authorisation, authentication
+
+Technical Story: [Allow agent admin role to be authenticated by Keycloak JWT | https://input-output.atlassian.net/browse/ATL-6074]
+
+## Context and Problem Statement
+
+External IAM authentication and authorization are currently limited to tenants using the wallets.
+Administrators currently rely on a static API key configured in the agent.
+Enhancing the security of admin access requires extending support for admin authentication through an external IAM.
+This introduces a concept of role to the authentication and authorization process.
+
+The existing tenant authentication and authorization model relies on UMA (user-managed access)
+which defines resources, policies, and permissions.
+Tenants gain access to resources based on defined permissions, effectively controlling resource access.
+In addition to wallet usage, the agent also handles wallet management,
+a functionality utilized by both tenants and administrators.
+While administrators don't directly use the wallet, they oversee its management.
+Integrating an auth model to distinguish between admins and tenants presents a new challenge.
```

Comment:
The mention of the `authorization model` is critical.
This ADR is about the Agent administrator authorization, but at the same time, it might be / should be a part of a larger model that includes a `manage` product (probably other products as well), and also be a part of more granular access to the Agent API (for instance, issuer/holder/verifier roles might have different access to the REST API)

Reply:
... I mean, the proposed authorization model must be extendable and built for longevity.

Reply:
Added a proposed authorization model to a decision outcome.